android/app/build.gradle

```diff
@@ -28,26 +27,21 @@ android {
         targetCompatibility JavaVersion.VERSION_1_8
     }
     kotlinOptions {
-        jvmTarget = '1.8'
+        jvmTarget = rootProject.kotlinJvmTargetVersion
     }
 }
 
 dependencies {
 
-    implementation 'androidx.core:core-ktx:1.6.0'
-    implementation 'androidx.appcompat:appcompat:1.3.1'
-    implementation 'com.google.android.material:material:1.4.0'
-    implementation 'androidx.constraintlayout:constraintlayout:2.1.0'
-    implementation project(path: ':library')
-
-    testImplementation 'junit:junit:4.13.2'
+    implementation "androidx.core:core-ktx:${rootProject.ktxVersion}"
+    implementation "androidx.appcompat:appcompat:${rootProject.appcompatVersion}"
+    implementation "com.google.android.material:material:${rootProject.materialVersion}"
+    implementation "androidx.constraintlayout:constraintlayout:${rootProject.constraintlayoutVersion}"
 
-    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
-}
+    implementation project(path: ':library')
 
+    testImplementation "junit:junit:${rootProject.junitVersion}"
 
-ktlint {
-    android.set(true)
-    outputColorName.set("RED")
+    androidTestImplementation "androidx.test.ext:junit:${rootProject.testExtJunitVersion}"
+    androidTestImplementation "androidx.test.espresso:espresso-core:${rootProject.espressoVersion}"
```

Comment:
is this a convention? what's the benefit of doing this

Reply:
in nodejs, we can use package manager to simply add/remove dependencies, like `pnpm add`. are we going to maintain the versions manually? or we can leverage some toolkit as well

Reply:
1. When you have lots of modules, build.gradle file can quickly get messy from dependencies.
Extract out library versions in gradle files make us managing dependency versions between different modules easier.
It will be more convenient when we upgrade dependencies.
We see this pattern in many projects.
ref:
[google-sunflower-project](https://github.com/android/sunflower/blob/91c4e0613e3e1a3c529235dc8ff73589a9047c79/build.gradle#L19-L53)

2. Early days, we need to download jar/arr files and add these files into the project. 
After that, gradle comes to our help and this is the way gradle manage dependencies.
Dependencies always have very long name and legacy problems. 
Managing them with some toolkit perhaps make no scese. 
Using gradle is the most efficient way today.